react/features/display-name/components/DisplayName.js

```diff
@@ -0,0 +1,237 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import { participantDisplayNameChanged } from '../../base/participants';
+
+/**
+ * React {@code Component} for displaying and editing a participant's name.
+ *
+ * @extends Component
+ */
+class DisplayName extends Component {
+    /**
+     * {@code DisplayName}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Whether or not the display name should be editable on click.
+         */
+        allowEditing: React.PropTypes.bool,
+
+        /**
+         * Invoked to update the participant's display name.
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * The participant's current display name.
+         */
+        displayName: React.PropTypes.string,
+
+        /**
+         * A string to append to the displayName, if provided.
+         */
+        displayNameSuffix: React.PropTypes.string,
+
+        /**
+         * The ID attribute to add to the component. Useful for global querying
+         * for the component by legacy components and torture tests.
+         */
+        idAttribute: React.PropTypes.string,
```

Comment:
I think generally you might want it to be just 'id' but I thought that might be confusing with participantID. I'll change it to elementId.